src/com/telesign/util/TeleSignRequest.java

```diff
@@ -370,7 +370,7 @@ public String getPostBody() {
 	 *			 error occurred while attempting to sign the Request.
 	 */
 	public String executeRequest() throws IOException {
-
+		setSigningMethod(auth);

```

Comment:
setSigningMethod(auth) gets 'x-ts-auth-method' added to TelesisgnRequest.headers and TelesisgnRequest.ts_headers fields.
As both 'headers & ts_headers' have data type as TreeMap, which means they cannot have duplicate key values, so old key-value will be overwitten by latest key value pair.

__Use case__: If the user calls setSigningMethod to set auth the method will add x-ts-auth-method but then we call it again during the executeRequest so it will add the header twice.
> Header won't be added twice as same value will be overwritten and map size will be 1.

__Test case__: There already exists a test case for this -- please refer to [shaMethodTest() from TeleSignRequestTest.java](https://github.com/TeleSign/java_telesign/blob/master/test/com/telesign/TeleSignRequestTest.java#L179)

Hope this helps.